example/driver/simpledriver.go

```diff
@@ -312,3 +313,11 @@ func (s *SimpleDriver) Discover() {
 	time.Sleep(time.Duration(s.serviceConfig.SimpleCustom.Writable.DiscoverSleepDurationSecs) * time.Second)
 	s.deviceCh <- res
 }
+
+func (s *SimpleDriver) ValidateDevice(device models.Device) error {
+	if _, ok := device.Protocols["invalid"]; ok {
+		return errors.New("invalid protocols")
+	}
```

Comment:
This should provide example of validating the actual expected protocol properties:

From https://github.com/edgexfoundry/device-sdk-go/blob/main/example/cmd/device-simple/res/devices/simple-device.toml
```
  [DeviceList.Protocols.other]
    Address = "simple01"
    Port = "300"
```

So it should:
1. Check that "other" exists
2. Check that "Address" exists and not empty
3. Check that "Port" exists, not empty and is a number

Reply:
updated.